tests/unit/client_tests.py

```diff
@@ -398,6 +399,17 @@ def test_create_existing_database(self):
         self.client.delete_database(dbname)
         self.client.disconnect()
 
+    def test_create_invalid_database_name(self):
+        """
+        Test creation of database with an invalid name
+        """
+        dbname = 'invalidDbName_'
+        self.client.connect()
+        with self.assertRaises(CloudantDatabaseException) as cm:
+            self.client.create_database(dbname, throw_on_exists=False)
+        self.assertEqual(cm.exception.status_code, 400)
+        self.client.disconnect()
+
```

Comment:
We probably should have an additional test that returns some other response code too. It could use a mock.